src/metatrain/experimental/phace/modules/cg.py

```diff
@@ -0,0 +1,131 @@
+from typing import Dict
+
+import numpy as np
+import torch
+import wigners
+
+
+def cgs_to_device_dtype(
+    cgs: Dict[str, Dict[str, torch.Tensor]], device: torch.device, dtype: torch.dtype
+):
+    cgs_device: Dict[str, Dict[str, torch.Tensor]] = {}
+    for key, value in cgs.items():
+        cgs_device[key] = {}
+        for k, v in value.items():
+            if k == "split_sizes":
+                cgs_device[key][k] = v
+            elif k == "C":
+                cgs_device[key][k] = v.to(device, dtype)
+            else:
+                cgs_device[key][k] = v.to(device)
+    return cgs_device
+
+
+def cg_combine_l1l2L(tensor12, cg_tensor):
+    out_tensor = tensor12 @ cg_tensor.reshape(
+        cg_tensor.shape[0] * cg_tensor.shape[1], cg_tensor.shape[2]
+    )
+    return out_tensor.swapaxes(1, 2)
+
+
+def get_cg_coefficients(l_max):
+    cg_object = ClebschGordanReal()
+    for l1 in range(l_max + 1):
+        for l2 in range(l_max + 1):
+            for L in range(abs(l1 - l2), min(l1 + l2, l_max) + 1):
+                cg_object._add(l1, l2, L)
+    return cg_object
+
+
+class ClebschGordanReal:
```

Comment:
we really need to put this in metatomic so we don't need to keep copy-pasting this code around